src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+        }
+
+        // Get from and to Strings
+        String inputInfo = input.substring(6);
+        String[] inputs1 = inputInfo.split(" /from ");
+        String taskName = inputs1[0];
+        String[] inputs2 = inputs1[1].split(" /to ");
+        String from = inputs2[0];
+        String to = inputs2[1];
+
+        // Error handling
+        if (taskName.length() == 0 || from.length() == 0 || to.length() == 0) {
+            throw new DukeException("EVENT DESCRIPTION AND FROM/TO DATE CANNOT BE EMPTY!");
+        }
+
+        Event newEvent = new Event(taskName, from, to);
+        System.out.println(horizontalLine);
+        System.out.println("OK. I'VE ADDED THIS TASK:");
+        System.out.println("[E][ ] " + newEvent);
+        taskArr.add(newEvent);
+        showCount();
+        System.out.println(horizontalLine);
+        updateListStorage();
+        readInput();
+    }
+
+    private static void deleteTask(String input) throws DukeException {
+        // Error handling
+        if (input.length() <= 7) {
+            throw new DukeException("DELETE NEEDS A TASK NUMBER!");
+        }
+
+        // Get delete task number
+        int index = Integer.parseInt(input.substring(7));
+
+        System.out.println(horizontalLine);
+        System.out.println("OK! I'VE DELETED THIS TASK: ");
+        System.out.println("" + (index) + ". " +
+                "[" + taskArr.get(index-1).getIcon() + "]" +
+                "[" + taskArr.get(index-1).getStatusIcon() + "] " +
+                taskArr.get(index-1));
+        taskArr.remove(index-1);
+        System.out.println(horizontalLine);
+        updateListStorage();
+        readInput();
+    }
+
+    private static void showMark(String input) {
+        int index = Integer.parseInt(input.substring(5));
+        System.out.println(horizontalLine);
+        System.out.println("I'VE MARKED THIS TASK AS DONE: ");
+        System.out.println("[X] " + taskArr.get(index-1));
+        taskArr.get(index-1).setDone(true);
+        System.out.println(horizontalLine);
+        updateListStorage();
+        readInput();
+    }
+
+    private static void showUnmark(String input) {
+        int index = Integer.parseInt(input.substring(7));
+        System.out.println(horizontalLine);
+        System.out.println("I'VE MARKED THIS TASK AS UNDONE: ");
+        System.out.println("[ ] " + taskArr.get(index-1));
+        taskArr.get(index-1).setDone(false);
+        System.out.println(horizontalLine);
+        updateListStorage();
+        readInput();
+    }
+
+    private static void showList() {
+        System.out.println(horizontalLine);
+        System.out.println("HERE ARE YOUR TASKS!");
+        for (int i = 0; i < taskArr.size(); i++) {
+            System.out.println("" + (i+1) + ". " +
+                    "[" + taskArr.get(i).getIcon() + "]" +
+                    "[" + taskArr.get(i).getStatusIcon() + "] " +
+                    taskArr.get(i));
```

Comment:
You can use `String.format` here if you want (and anywhere else with a lot of concatenation)

Also, since you're using `taskArr.get(i)` a lot, maybe you can store it in a variable.
```suggestion
            Task task = taskArr.get(i);
            System.out.println(String.format(
                    "%d. [%s] [%s] %s",
                    i + 1, task.getIcon(), task.getStatusIcon(), task));
```